core/src/main/java/org/fao/geonet/kernel/DataManager.java

```diff
@@ -27,3352 +27,656 @@
-     */
-    public void validate(String schema, Element md) throws Exception {
-        String schemaLoc = md.getAttributeValue("schemaLocation", Namespaces.XSI);
-        if (Log.isDebugEnabled(Geonet.DATA_MANAGER))
-            Log.debug(Geonet.DATA_MANAGER, "Extracted schemaLocation of " + schemaLoc);
-        if (schemaLoc == null) schemaLoc = "";
-
-        if (schema == null) {
-            // must use schemaLocation
-            Xml.validate(md);
-        } else {
-            // if schemaLocation use that
-            if (!schemaLoc.equals("")) {
-                Xml.validate(md);
-                // otherwise use supplied schema name
-            } else {
-                Xml.validate(getSchemaDir(schema).resolve(Geonet.File.SCHEMA), md);
+                Log.debug(Geonet.DATA_MANAGER, "Hopefully this is cron job or routinely background task. Who called us?",
+                        new Exception("Dummy Exception to know the stacktrace"));
+            } catch (Exception e) {
+                // Silent. This is just to log the stacktrace here
             }
         }
     }
 
-    /**
-     * TODO javadoc.
-     */
-    public Element validateInfo(String schema, Element md, ErrorHandler eh) throws Exception {
-        String schemaLoc = md.getAttributeValue("schemaLocation", Namespaces.XSI);
-        if (Log.isDebugEnabled(Geonet.DATA_MANAGER))
-            Log.debug(Geonet.DATA_MANAGER, "Extracted schemaLocation of " + schemaLoc);
-        if (schemaLoc == null) schemaLoc = "";
-
-        if (schema == null) {
-            // must use schemaLocation
-            return Xml.validateInfo(md, eh);
-        } else {
-            // if schemaLocation use that
-            if (!schemaLoc.equals("")) {
-                return Xml.validateInfo(md, eh);
-                // otherwise use supplied schema name
-            } else {
-                return Xml.validateInfo(getSchemaDir(schema).resolve(Geonet.File.SCHEMA), md, eh);
-            }
-        }
+    @Deprecated
+    public static void validateMetadata(String schema, Element xml, ServiceContext context) throws Exception {
+        GeonetContext gc = (GeonetContext) context.getHandlerContext(Geonet.CONTEXT_NAME);
+        gc.getBean(IMetadataValidator.class).validateMetadata(schema, xml, context, " ");
     }
 
-    /**
-     * Creates XML schematron report.
-     */
-    public Element doSchemaTronForEditor(String schema, Element md, String lang) throws Exception {
-        // enumerate the metadata xml so that we can report any problems found
-        // by the schematron_xml script to the geonetwork editor
-        editLib.enumerateTree(md);
-
-        // get an xml version of the schematron errors and return for error display
-        Element schemaTronXmlReport = getSchemaTronXmlReport(schema, md, lang, null);
-
-        // remove editing info added by enumerateTree
-        editLib.removeEditingInfo(md);
-
-        return schemaTronXmlReport;
+    @Deprecated
```

Comment:
Has been change the code using this code to use the new code? Or this will be part of other pull request?

Reply:
I haven't changed anything. DataManager is deprecated, but still works (using the helper classes, which contains the original source code).

In the future it will be good to completely remove DataManager for cleanliness, but it will work as it worked until now.